types/react-dom/index.d.ts

```diff
@@ -29,7 +29,7 @@ export function unmountComponentAtNode(container: Element | DocumentFragment): b
 export function createPortal(
     children: ReactNode,
     container: Element | DocumentFragment,
-    key?: null | string,
+    key?: null | string | number,
```

Comment:
this can actually just be 
```suggestion
    key?: React.Key,
```

Reply:
@eps1lon 
OK, but should I remove `null`, or leave it as is? Removing `null` might cause a breaking change.

```tsx
export function createPortal(
    children: ReactNode,
    container: Element | DocumentFragment,
    key?: Key | null,
): ReactPortal;
```

Reply:
Good catch, keep the `null`

Reply:
OK, now fixed.